pass correct object type to association reflection

formigarafa opened the pull request:
This is to make the association relation receive the expected parameter
(a AR instance, not its class). fix #952

mcmire commented on spec/unit/shoulda/matchers/active_record/association_matchers/model_reflection_spec.rb:
```diff
@@ -92,13 +92,15 @@
-              belongs_to :country, -> { where(mood: 'nice') }
+            columns = { country_id: :integer, spirit: :string }
+            person_model = define_model(:person, columns) do
+              belongs_to :country, ->(person) { where(mood: person.spirit) }
```
Thanks for this PR. Instead of modifying this test (which is a known use case) what do you think about making a new test?

formigarafa replied:
I also thought the same, previously I committed as such, but it looked so much as a duplication that I removed. I will put it back.

formigarafa commented:
I wrote another test before but it was duplicating most of the previous code, so I removed it.
Now I have fixed it to only test the change. It should be better, now.

mcmire commented:
Sorry for the delay in getting to this. I took the liberty of adding a missing test to this and cleaning up a couple things around the existing tests. This is merged now!

formigarafa commented:
@mcmire , I don't mind at all and I really appreciate that. Thanks.
